plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchExecutor.groovy

```diff
@@ -265,6 +271,28 @@ class AwsBatchExecutor extends Executor implements ExtensionPoint {
     @PackageScope
     ThrottlingExecutor getReaper() { reaper }
 
+    void killTask(String jobId) {
+        // extract job array id
+        if( jobId.contains(':') )
+            jobId = jobId.split(':')[0]
+
+        // prevent duplicate delete requests on the same job
+        if( jobId in deletedJobs )
+            return
+        else
+            deletedJobs.add(jobId)
```

Comment:
Here a race condition can happen

Reply:
In theory maybe, but in practice only the task monitor kills jobs one at a time

Reply:
It was cheap to make thread-safe